app/channel.go

```diff
@@ -1329,6 +1329,21 @@ func (a *App) GetChannelMembersTimezones(channelId string) ([]string, *model.App
 	return model.RemoveDuplicateStrings(timezones), nil
 }
 
+func (a *App) GetChannelMembersEmails(channelId string) ([]string, *model.AppError) {
+	result := <-a.Srv.Store.Channel().GetChannelMembersEmails(channelId)
```

Comment:
We can use a.Srv.Store.User().GetAllProfilesInChannel(channel.Id, false)  and parse for email and return.

Reply:
Hi @amitmall, 

I have used the GetAllProfilesInChannel method with cache true and removed the store method which I had created. Please review the changes.

https://github.com/nykashish/mattermost-server/commit/bd0adcd571bba195e9e7816c866da907c1ba8100